api/jobset/v1alpha2/pod_webhook.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// +kubebuilder:webhook:path=/mutate-v1-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=mpod.kb.io,sideEffects=None,admissionReviewVersions=v1
```

Comment:
It can be added via the kubebuilder marker? I found a reference to `objectSelector` here https://book.kubebuilder.io/cronjob-tutorial/running-webhook but it links to the `ObjectSelector` field in the k8s admission `ValidatingWebhook` struct which doesn't seem like what we need, since we are using the `controller-runtime` function `NewWebhookManagedBy(...)` to build our webhooks.

Can you clarify how you suggest adding the objectSelector here?

Reply:
I think we can add it to the webhooks manifests.yaml directly, I left a comment there.

Reply:
the controller-runtime is on the client side of the equation, so it will not be where this will be setup. We need it setup on the api-server side, which is what the manifests.yaml file is about